carbonserver/carbonserver/api/infra/repositories/repository_runs.py

```diff
@@ -1,59 +1,146 @@
-from sqlalchemy import exc
-from sqlalchemy.orm import Session
+import uuid
+from contextlib import AbstractContextManager
+from typing import List
+
+from dependency_injector.providers import Callable
 
 from carbonserver.api import schemas
 from carbonserver.api.domain.runs import Runs
-from carbonserver.api.errors import DBError, DBErrorEnum, DBException
-from carbonserver.database import sql_models
+from carbonserver.api.schemas import Run, RunCreate
+from carbonserver.database.sql_models import Run as SqlModelRun
 
 """
-Here there is all the method to manipulate the experiment data
+Here there is all the methods to manipulate the run data
 """
 
 
 class SqlAlchemyRepository(Runs):
-    def __init__(self, db: Session):
-        self.db = db
+    def __init__(self, session_factory) -> Callable[..., AbstractContextManager]:
+        self.session_factory = session_factory
 
-    def add_run(self, run: schemas.RunCreate) -> sql_models.Run:
+    def add_run(self, run: RunCreate) -> Run:
         """Save an Run to the database.
         :run: An Run in pyDantic BaseModel format.
         :returns: An Run in SQLAlchemy Model format.
         :rtype: models.Run
         """
-        db_run = sql_models.Run(
-            timestamp=run.timestamp,
-            experiment_id=run.experiment_id,
-        )
-        try:
-            self.db.add(db_run)
-            self.db.commit()
-            self.db.refresh(db_run)
-            return db_run
-        except exc.IntegrityError as e:
-            # Sample error : sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "emissions" violates foreign key constraint "fk_emissions_runs"
-            self.db.rollback()
-            raise DBException(
-                error=DBError(code=DBErrorEnum.INTEGRITY_ERROR, message=e.orig.args[0])
-            )
-        except exc.DataError as e:
-            self.db.rollback()
-            # Sample error :  sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type uuid: "5050f55-406d-495d-830e-4fd12c656bd1"
-            raise DBException(
-                error=DBError(code=DBErrorEnum.DATA_ERROR, message=e.orig.args[0])
+        with self.session_factory() as session:
+            db_run = SqlModelRun(
+                id=uuid.uuid4(),
+                timestamp=run.timestamp,
+                experiment_id=run.experiment_id,
             )
+            session.add(db_run)
+            session.commit()
+            session.refresh(db_run)
+            return self.map_sql_to_schema(db_run)
+
+        # db_run = sql_models.Run(
+        #     timestamp=run.timestamp,
```

Comment:
All commented code can be removed. If needed in the future we will rewrite it easier (and it will always be available in the history).